core/services/pipeline/task.merge.go

```diff
@@ -0,0 +1,51 @@
+package pipeline
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+)
+
+//
+// Return types:
+//     map[string]interface{}
+//
+type MergeTask struct {
+	BaseTask `mapstructure:",squash"`
+	Left     string `json:"left"`
+	Right    string `json:"right"`
+}
+
+var _ Task = (*MergeTask)(nil)
+
+func (t *MergeTask) Type() TaskType {
+	return TaskTypeMergeTask
+}
+
+func (t *MergeTask) Run(_ context.Context, vars Vars, inputs []Result) (result Result, runInfo RunInfo) {
+	_, err := CheckInputs(inputs, 0, 1, 0)
+	if err != nil {
+		return Result{Error: errors.Wrap(err, "task inputs")}, runInfo
+	}
+
+	var (
+		lMap MapParam
+		rMap MapParam
+	)
+	err = multierr.Combine(
+		errors.Wrap(ResolveParam(&lMap, From(VarExpr(t.Left, vars), NonemptyString(t.Left), Input(inputs, 0))), "left-side"),
```

Comment:
so here the `Input(inputs, 0)` parts merges 'inputs' into 'left' ?

Reply:
No it takes the param `left` if specified otherwise falls back to the input